### PR TITLE
add %java

### PR DESCRIPTION
**Pull request by timothyclemansinsea**

### CODE REVIEW

File: src/smc_sagews/smc_sagews/sage_salvus.py [hunk 1]
```diff
@@ -3731,7 +3731,64 @@ def go(s):
...
+        }
+
+    You can set the whole worksheet to be in java mode by typing
+
+        %default_mode java
 
+    NOTE:
+
+    - There is no relation between one cell and the next.  Each is a separate
+      self-contained go program, which gets compiled and run, with the only
```
Comment (williamstein) [inline src/smc_sagews/smc_sagews/sage_salvus.py]: go --> java

File: src/smc_sagews/smc_sagews/sage_salvus.py [hunk 2]
```diff
@@ -3731,7 +3731,64 @@ def go(s):
...
+# Java mode
+########################################################
+def java(s):
+    """
+    Run a Java program.  For example,
+
+        %java
+        public class YourName {
+            public static void main(String[] args) {
+
```
Comment (williamstein) [inline src/smc_sagews/smc_sagews/sage_salvus.py]: maybe too much whitespace?

File: src/smc_sagews/smc_sagews/sage_salvus.py [hunk 3]
```diff
@@ -3731,7 +3731,64 @@ def go(s):
...
+        sys.stdout.flush()
+        sys.stderr.flush()
+        if not os.path.exists(name+'.class'): # failed to produce executable
+            return
+        (child_stdin, child_stdout, child_stderr) = os.popen3('java %s'%name)
+        sys.stdout.write(child_stdout.read())
+        sys.stderr.write('\n'+child_stderr.read())
+        sys.stdout.flush()
+        sys.stderr.flush()
+    finally:
```
Comment (williamstein) [inline src/smc_sagews/smc_sagews/sage_salvus.py]: When I tried this there was also some red in the output -- that's kind of weird, right?

Comment (timothyclemansinsea) [inline src/smc_sagews/smc_sagews/sage_salvus.py]: @williamstein I'm not sure what the right thing to do about that is. I assume it is stderr

Comment (williamstein) [inline src/smc_sagews/smc_sagews/sage_salvus.py]: It is stderr.  I'm not sure either.   We can leave it as is for now, but it is annoying, right?

Comment (timothyclemansinsea) [inline src/smc_sagews/smc_sagews/sage_salvus.py]: @williamstein yep it is annoying. Maybe ignore those particular lines?

**Pull request merged**